sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -95,4 +101,94 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       return item;
     });
   }
+
+  /**
+   * This is to fetch the Education Offering
+   * that are associated with the Location, Program
+   * and offering
+   * @param locationId
+   * @param programId
+   * @param offeringId
+   * @returns
+   */
+  async getProgramOffering(
+    locationId: number,
+    programId: number,
+    offeringId: number,
+  ): Promise<ProgramOfferingModel> {
+    return this.repo
+      .createQueryBuilder("offerings")
+      .select([
+        "offerings.id",
+        "offerings.name",
+        "offerings.studyStartDate",
+        "offerings.studyEndDate",
+        "offerings.breakStartDate",
+        "offerings.breakEndDate",
+        "offerings.actualTuitionCosts",
+        "offerings.programRelatedCosts",
+        "offerings.mandatoryFees",
+        "offerings.exceptionalExpenses",
+        "offerings.tuitionRemittanceRequestedAmount",
+        "offerings.offeringDelivered",
+        "offerings.lacksStudyDates",
+        "offerings.lacksStudyBreaks",
+        "offerings.lacksFixedCosts",
+        "offerings.tuitionRemittanceRequested",
+      ])
+      .innerJoin("offerings.educationProgram", "educationProgram")
+      .innerJoin("offerings.institutionLocation", "institutionLocation")
+      .where(
```

Comment:
Just a recommendation, the andWhere could be used to break the 3 where condition into 3 lines.

Reply:
done